src/inference/dev_api/threading/ie_thread_safe_containers.hpp

```diff
@@ -48,6 +49,31 @@ template <typename T>
 using ThreadSafeQueue = tbb::concurrent_queue<T>;
 template <typename T>
 using ThreadSafeBoundedQueue = tbb::concurrent_bounded_queue<T>;
+template <typename T>
+class ThreadSafeBoundedPriorityQueue {
+public:
+    ThreadSafeBoundedPriorityQueue() = default;
+    bool try_push(T value) {
+        _mutex.lock();
```

Comment:
maybe use unique lock，if(_capacity==0) always lock

Reply:
thanks for pointing out this bug, unique lock maybe too heavy for this piece of code, *push* itself is thread safe, fix it by adding unlock.

Reply:
pls use the tbb::concurrent_bounded_queue

Reply:
would making the _capacity atomic do the trick in a simpler way?